modules/docs/arrow-docs/docs/docs/datatypes/listk/README.md

```diff
@@ -6,4 +6,89 @@ permalink: /docs/datatypes/listk/
 
 ## ListK
 
-TODO. Meanwhile you can find a short description in the [intro to datatypes]({{ '/docs/datatypes/intro/' | relative_url }}).
+ListK wraps over the platform `List` type to make it a (type constructor)[/docs/patterns/glossary/#type-constructors].
+
+It can be created from Kotlin Sequence type with a convenient `k()` function.
+
+```kotlin:ank
+import arrow.*
+import arrow.data.*
+
+listOf(1, 2, 3).k()
+```
+
+and unwrapped with the field `list`.
+
+```kotlin:ank
+listOf(1, 2, 3).k().list
+```
+
+ListK implements operators from many useful typeclasses.
+
+For instance, it has `combineK` from the [`SemigroupK`](/docs/typeclasses/semigroupk/) typeclass.
+
+It can be used to cheaply combine two lists:
+
+```kotlin:ank
+val hello = listOf('h', 'e', 'l', 'l', 'o').k()
+val commaSpace = listOf(',', ' ').k()
+val world = listOf('w', 'o', 'r', 'l', 'd').k()
+
+hello.combineK(commaSpace.combineK(world)) == hello.combineK(commaSpace).combineK(world)
+```
+
+The functions `traverse` and `sequence` come from [`Traverse`](/docs/typeclasses/traverse/).
+
+Traversing a list creates a new container (`Kind<F, A>`)[/docs/patterns/glossary/#type-constructors] by combining the result of a function applied to each element:
+
+```kotlin:ank
+val numbers =  listOf(Math.random(), Math.random(), Math.random()).k()
+requests.traverse(Option.applicative()) { if (it > 0.5) Some(it) else None }
+```
+
+and complements the convenient function `sequence()` that converts a list of `List<Kind<F, A>>` into a `Kind<F, List<A>>`:
+
+```kotlin:ank
+fun request() = Some(Math.random())
+
+val requests =  listOf(request(), request(), request()).k()
+requests.sequence(Option.applicative()).fix()
+```
+
+A more general approach is aggregating values using `foldLeft` and `foldRight` from [`Foldable`](/docs/typeclasses/foldable).
```

Comment:
Not sure I understand this sentence right. Should it be?

> A more general approach **for** aggregating values?

Reply:
I'll rewrite.